src/SignalR/common/Shared/ReflectionHelper.cs

```diff
@@ -28,6 +32,9 @@ public static bool IsStreamingType([DynamicallyAccessedMembers(DynamicallyAccess
         {
             if (nullableType.IsGenericType && nullableType.GetGenericTypeDefinition() == typeof(ChannelReader<>))
             {
+                Debug.Assert(nullableType.GetGenericArguments().Length == 1);
+
+                _streamTypeLookup.GetOrAdd(type, nullableType.GetGenericArguments()[0]);
```

Comment:
If we keep the cache, we should use it for repeated calls to `IsStreamingType` itself. And we should be careful to not use the cache for different values of `mustBeDirectType`. Maybe we should only cache in the `false` case.